### PR TITLE
Streamline Inheritance + dependsOnMethods

### LINKED ISSUE #2658: Inheritance + dependsOnMethods
### Test case sample
```java
public abstract class Base {
    @Test
    void test() {
        assertEquals(getClass().getSimpleName(), "Test1");
    }
}

class Test1 extends Base {
    @Test(dependsOnMethods = "test")
    void otherTest() {}
}

class Test2 extends Base {
    @Test(dependsOnMethods = "test")
    void otherTest() {}
}
```

### Expected behavior
image

### Actual behavior
image

### TestNG Version
7.4.0
I do not know if it worked in the past

### Is the issue reproducible on runner?

- [ ] Shell
- [ ] Maven
- [X] Gradle
- [ ] Ant
- [ ] Eclipse
- [X] IntelliJ
- [ ] NetBeans

I do not understand if actual behavior is intended or is a bug, but would that be ok to change it? if not, is there any option that I could override it?

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #2658

Refactored logic to group test results based on 
Instances depending upon whether 

1. Test has group dependency (or)
2. Test has method dependency

Also refactored the tests to move away from different
Base class flavours into converging on SimpleBaseTest

Fixes #2658  .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`
- [X] Auto applied styling via `./gradlew autostyleApply`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

**Note:** For more information on contribution guidelines  please make sure you refer our [Contributing](.github/CONTRIBUTING.md) section for detailed set of steps.

### CODE REVIEW

File: testng-core/src/test/java/test/dependent/DependentTest.java [hunk 1]
```diff
@@ -227,4 +206,84 @@ public void simpleCyclingDependencyShouldWorkWitParallelism(
                 + Arrays.deepToString(runMethods));
     assertThat(listener.getSucceedMethodNames()).containsExactlyInAnyOrder(runMethods[0]);
   }
+
+  @Test(description = "GITHUB-2658")
+  public void testMethodDependencyAmidstInheritance() {
+    TestNG testng = new TestNG();
```
Comment (juherr) [inline testng-core/src/test/java/test/dependent/DependentTest.java:212]: Why not using SimpleBaseTest.create here?

File: testng-core/src/test/java/test/dependent/DependentTest.java [hunk 2]
```diff
@@ -227,4 +206,84 @@ public void simpleCyclingDependencyShouldWorkWitParallelism(
...
+    testng.setTestClasses(new Class[] {PassingClassSample.class, FailingClassSample.class});
+    testng.setVerbose(2);
```
Comment (juherr) [inline testng-core/src/test/java/test/dependent/DependentTest.java:214]: Is verbose important for the test?

File: testng-core/src/test/java/test/dependent/DependentTest.java (hunk 1 above)
Comment (krmahadevan) [reply testng-core/src/test/java/test/dependent/DependentTest.java:212]: That was a copy paste error. Fixed.

File: testng-core/src/test/java/test/dependent/DependentTest.java (hunk 2 above)
Comment (krmahadevan) [reply testng-core/src/test/java/test/dependent/DependentTest.java]: Nah. Its not. It came in as part of a template `IssueTest` i use when I first build the unit tests :) Forgot to get rid of it, after I moved that test into this class. Basically a bone headed mistake 😁 
Fixed.

**Comment by krmahadevan** [comment]
@juherr - Please let me know if this can now be merged.

**Review by juherr — APPROVED** [review]

**Pull request merged**